src/navsat_transform.cpp

```diff
@@ -867,42 +868,44 @@ namespace RobotLocalization
-
-      // Rotate the covariance as well
-      tf2::Matrix3x3 rot(cartesian_world_transform_.getRotation());
-      Eigen::MatrixXd rot_6d(POSE_SIZE, POSE_SIZE);
-      rot_6d.setIdentity();
-
-      for (size_t rInd = 0; rInd < POSITION_SIZE; ++rInd)
+      bool is_transformed = getRobotOriginWorldPose(transformed_cartesian_gps, transformed_cartesian_robot, gps_odom.header.stamp);
+      if (is_transformed)
       {
-        rot_6d(rInd, 0) = rot.getRow(rInd).getX();
-        rot_6d(rInd, 1) = rot.getRow(rInd).getY();
-        rot_6d(rInd, 2) = rot.getRow(rInd).getZ();
-        rot_6d(rInd+POSITION_SIZE, 3) = rot.getRow(rInd).getX();
-        rot_6d(rInd+POSITION_SIZE, 4) = rot.getRow(rInd).getY();
-        rot_6d(rInd+POSITION_SIZE, 5) = rot.getRow(rInd).getZ();
-      }
+        // Rotate the covariance as well
+        tf2::Matrix3x3 rot(cartesian_world_transform_.getRotation());
+        Eigen::MatrixXd rot_6d(POSE_SIZE, POSE_SIZE);
+        rot_6d.setIdentity();
 
-      // Rotate the covariance
-      latest_cartesian_covariance_ = rot_6d * latest_cartesian_covariance_.eval() * rot_6d.transpose();
+        for (size_t rInd = 0; rInd < POSITION_SIZE; ++rInd)
+        {
+          rot_6d(rInd, 0) = rot.getRow(rInd).getX();
+          rot_6d(rInd, 1) = rot.getRow(rInd).getY();
+          rot_6d(rInd, 2) = rot.getRow(rInd).getZ();
+          rot_6d(rInd+POSITION_SIZE, 3) = rot.getRow(rInd).getX();
+          rot_6d(rInd+POSITION_SIZE, 4) = rot.getRow(rInd).getY();
+          rot_6d(rInd+POSITION_SIZE, 5) = rot.getRow(rInd).getZ();
+        }
 
-      // Now fill out the message. Set the orientation to the identity.
-      tf2::toMsg(transformed_cartesian_robot, gps_odom.pose.pose);
-      gps_odom.pose.pose.position.z = (zero_altitude_ ? 0.0 : gps_odom.pose.pose.position.z);
+        // Rotate the covariance
+        latest_cartesian_covariance_ = rot_6d * latest_cartesian_covariance_.eval() * rot_6d.transpose();
 
-      // Copy the measurement's covariance matrix so that we can rotate it later
-      for (size_t i = 0; i < POSE_SIZE; i++)
-      {
-        for (size_t j = 0; j < POSE_SIZE; j++)
+        // Now fill out the message. Set the orientation to the identity.
+        tf2::toMsg(transformed_cartesian_robot, gps_odom.pose.pose);
+        gps_odom.pose.pose.position.z = (zero_altitude_ ? 0.0 : gps_odom.pose.pose.position.z);
+
+        // Copy the measurement's covariance matrix so that we can rotate it later
+        for (size_t i = 0; i < POSE_SIZE; i++)
         {
-          gps_odom.pose.covariance[POSE_SIZE * i + j] = latest_cartesian_covariance_(i, j);
+          for (size_t j = 0; j < POSE_SIZE; j++)
+          {
+            gps_odom.pose.covariance[POSE_SIZE * i + j] = latest_cartesian_covariance_(i, j);
+          }
         }
-      }
 
-      // Mark this GPS as used
-      gps_updated_ = false;
-      new_data = true;
+        // Mark this GPS as used
+        gps_updated_ = false;
```

Comment:
Do we need to do something with this class-member variable in an `else` condition? If `gps_updated_` remains true, what consequence does that have?

Reply:
> If gps_updated_ remains true, what consequence does that have?

The only consequence is that next time the `prepareGpsOdometry(..)` is called , then it would use the latest `latest_cartesian_pose_` ( made from the latest time a gps fix message was received.) . We'd also be using that latest gps fix header time as well.